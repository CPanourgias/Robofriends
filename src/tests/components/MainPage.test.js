import { shallow } from "enzyme";
import MainPage from "../../components/MainPage";
import React from "react";

describe('Main Page tests', () => {


    // beforeAll(() => {
    //     /* Runs before all tests */
    // })
    // afterAll(() => {
    //     /* Runs after all tests */
    // })
    // afterEach(() => {
    //     /* Runs after each test */
    // })

    let wrapper;
    beforeEach(() => {
        const mockProps = {
            onRequestRobots: jest.fn(),
            robots: [{
                id: 1,
                name: 'John Snow',
                email: 'john@snow.com'
            }],
            searchField: '',
            isPending: false
        }
        wrapper = shallow(<MainPage { ...mockProps }/>);
    });

    it('renders MainPage', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('filters robots correctly', () => {

        let filteredRobots = [{
            id: 1,
            name: 'John Snow',
            email: 'john@snow.com'
        }];

        // No input
        expect(wrapper.instance().filteredRobots()).toEqual(filteredRobots);

        // Correct input
        wrapper.setProps( { searchField: 'John' } );
        expect(wrapper.instance().filteredRobots()).toEqual(filteredRobots);

        // Incorrect input
        filteredRobots = [];
        wrapper.setProps( { searchField: 'a' } );
        expect(wrapper.instance().filteredRobots()).toEqual(filteredRobots);
    });
});