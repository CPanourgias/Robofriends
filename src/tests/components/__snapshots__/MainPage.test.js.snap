// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page tests renders MainPage 1`] = `ShallowWrapper {}`;
